File: factory/src/main/java/com/zuzex/factory/repository/OrderRepository.java
package com.zuzex.factory.repository;

import com.zuzex.common.model.Status;
import com.zuzex.factory.model.Order;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OrderRepository extends JpaRepository<Order, Long> {

    Optional<Order> findByIdAndStatus(Long id, Status status);
}

File: car-factory/src/main/kotlin/com/zuzex/carfactory/repository/OrderRepository.kt
package com.zuzex.carfactory.repository

import com.zuzex.carfactory.model.Order
import com.zuzex.common.model.Status
import org.springframework.data.jpa.repository.JpaRepository

interface OrderRepository : JpaRepository<Order, Long> {

    fun findByIdAndStatus(id: Long, status: Status): Order?
}
